bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/handler/SimpleDeviceHandler.java

```diff
@@ -0,0 +1,251 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solarwatt.internal.handler;
+
+import static org.openhab.binding.solarwatt.internal.SolarwattBindingConstants.*;
+
+import java.text.MessageFormat;
+import java.util.Map;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.solarwatt.internal.channel.SolarwattChannelTypeProvider;
+import org.openhab.binding.solarwatt.internal.domain.SolarwattChannel;
+import org.openhab.binding.solarwatt.internal.domain.model.Device;
+import org.openhab.core.library.CoreItemFactory;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.*;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.util.UnitUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SimpleDeviceHandler} bundles everything related to generic talking to devices.
+ *
+ * @author Sven Carstens - Initial contribution
+ */
+@NonNullByDefault
+public class SimpleDeviceHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(SimpleDeviceHandler.class);
+
+    private final SolarwattChannelTypeProvider channelTypeProvider;
+
+    public SimpleDeviceHandler(Thing thing, SolarwattChannelTypeProvider channelTypeProvider) {
+        super(thing);
+        this.channelTypeProvider = channelTypeProvider;
+    }
+
+    /**
+     * Bring the thing online and update state from the bridge.
+     */
+    @Override
+    public void initialize() {
+        final EnergyManagerHandler bridgeHandler = this.getEnergyManagerHandler();
+        if (bridgeHandler != null) {
+            this.initDeviceChannels();
+            this.updateDeviceProperties();
+            this.updateDeviceChannels();
+        } else {
+            this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Received null bridge while initializing!");
+        }
+    }
+
+    /**
+     * Process the command for this thing.
+     *
+     * Only refresh is supported in this case.
+     *
+     * @param channelUID channel for which the command was issued
+     * @param command to execute
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        final EnergyManagerHandler bridgeHandler = this.getEnergyManagerHandler();
+        if (bridgeHandler != null) {
+            if (command instanceof RefreshType) {
+                this.logger.debug("update command for {}", this.getThing().getUID());
```

Comment:
This is a double up, as all commands are already logged to the event.log so you dont need to log it again causing higher load even if debug is not enabled it still will cause a load when it checks.

Reply:
Excessive logging is removed.